pkg/abaputils/addonDescriptor_test.go

```diff
@@ -0,0 +1,160 @@
+package abaputils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func readFileMock(FileName string) ([]byte, error) {
+	return []byte(FileName), nil
+}
+
+func TestAddonDescriptorNew(t *testing.T) {
+	t.Run("Import addon.yml", func(t *testing.T) {
+		var addonDescriptor AddonDescriptor
+		err := addonDescriptor.initFromYmlFile(TestAddonDescriptorYAML, readFileMock)
+		CheckAddonDescriptorForRepositories(addonDescriptor)
+
+		assert.NoError(t, err)
+		assert.Equal(t, "/DMO/myAddonProduct", addonDescriptor.AddonProduct)
+		assert.Equal(t, "/DMO/REPO_A", addonDescriptor.Repositories[0].Name)
+		assert.Equal(t, "JEK8S273S", addonDescriptor.Repositories[1].CommitID)
+		assert.Equal(t, "FR", addonDescriptor.Repositories[1].Languages[2])
+		assert.Equal(t, `ISO-DEENFR`, addonDescriptor.Repositories[1].GetAakAasLanguageVector())
+	})
+}
+
+var TestAddonDescriptorYAML = `---
```

Comment:
Wouldn't it be better to define this var within the test function?

Reply:
I usually like to have such 'huge' junks of data outside for readability and reuse, is there some guideline to place it inside the test function?

Reply:
Locally I just reused the new data in the old test (below the variable) and found the obsolete attributes:
addonUniqueId: myAddonId
customerID: 1234
Should we remove them from the structure? They are not used in our coding and also not mentiond in the documentation as far as I could see

Reply:
I think
```
addonUniqueId: myAddonId
customerID: 1234
```
were added to test if parameters that are not part of the expected YML file are handled without errors (by simply ignoring them)